### ISSUE TITLE
Array copy item with UUID: first copy has a duplicated UUID

### ISSUE THREAD

**Issue by simontaurus** [json-editor/json-editor#1481]
### General information

- json-editor version: 2.14.0

### Expected behavior

Each array element should have a unique ID

### Actual behavior

The first copy has the same UUID as the first element
Seems like #1129 is only partially solved in the current release

### Steps to reproduce the behavior

Direct link to example: [playground](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUBXRBTA+vAhgLa75RQBOMARulFsigGb4A22ANCBOWIdDlSyMw5XABMsLLHVTM2WTpnFD86FlBwA3VunqzWHEIXwAPHBOgALVAAZONMghyIopXITATUARk4BjfGxnLCQYWE1cbkgscigAT2D8cj9rJgMFEERLMAB3HBiecgYQGHg6cnw/WAQQThg/BBZqVBBhMvwc+l4sAFZa8CoAKywqnBZ8OLBaFvgRYxZ+xBSsY1RQeIgsFrAhkah+qM3YmD00EHR0GDE1kA2tlEyKUoBzfuFyY32Hi6v+2CgpC1EC8pAACH7XTiiACOl1E1xQFF0nAkzDUXxAVHwBAIIAAvopLmIAILkCpxG53FpJcl/MKA75E0E0ib9GFwrAIpEZMIrBjrOKbba7KoHHhHWCnDBEymC+6PSjwV6cd6fFoQ/F4rVazhQSwreVUMBgKAuCoQAAsbKwsJg8IECVRqnU+nkKJgiAg4wS7LtnJwjQpaTdmWyeUgGlKrsMsxw+DEYjCcAILBwhxikoYckM+EGpmjGUTiHwVCk+UTGkGiBqwcMRZLZYaLHGEGwBfdxdLkXFGZOWfSHYbJDJEzjCfbIHrXbjI59WBEEnIE6nZZZjsk0nu2YyISHM/JOAaECD3M4a8PCAoYDYOCgkGXHr354kUjocebOB4OX7IZXw4PL6bmMgQaF+BZakAA===)

```javascript
{
  "type": "object",
  "properties": {
    "uuid": {
      "type": "string",
      "format": "uuid",
      "title": "single uuid",
      "required": true,
      "default": "banana"
    },
    "uuidArray": {
      "type": "array",
      "title": "uuid array",
      "required": true,
      "items": {
        "type": "object",
        "properties": {
          "uuid": {
            "type": "string",
            "format": "uuid"
          }
        }
      }
    }
  }
}
```

Creating a first element a two copies leads to 

```javascript
{
  "uuid": "2df6ccc8-b03e-4922-bdba-ad83bc458e4e",
  "uuidArray": [
    {
      "uuid": "1c0ed826-8b36-4639-b909-381f7ab5a140"
    },
    {
      "uuid": "6492984d-8013-4b57-863f-fff7ea01a70c"
    },
    {
      "uuid": "6492984d-8013-4b57-863f-fff7ea01a70c"
    }
  ]
}
```

A workaround is to use a uuid template instead: #1485

**Comment by germanbisurgi** [comment]
Hi @simontaurus,

When adding uuid items with the "+ Item" button everything works like expected. Because the uuid editor create their own value at creation in order to be universally unique.

I guess the issue it's about setting the array value directly in which 2 items are identical. If that is so the value should be stripped off from duplicated items before setting the value of the array. Additionally the constrain "uniqueItems" can be added to the array schema to show validation errors.

I'm closing this because i think that was the issue, which is not a json editor bug, but feel free to reopen it if i missed something :)

**Comment by simontaurus** [comment]
> When copying items the Array editor will check if it#s item is type string and has format uuid. In that case the uuid will be regenerated. In this case it does not override uuid because it#s items are of type object
Makes sense, so the copied item is a true and full copy, including the uuid. One could argue that a UUID is per definition global unique and should never be duplicated, but with a working `copyRow`callback (see #1483) it's maybe best to enforce the uuid regeneration there
